src/api/posts/src/routes/posts.js

```diff
@@ -60,14 +60,31 @@ posts.get('/', validatePostsQuery(), async (req, res, next) => {
     first: `/posts?per_page=${perPage}&page=${1}`,
     last: `/posts?per_page=${perPage}&page=${Math.floor(postsCount / perPage)}`,
   });
-  res.json(
-    ids
+  let data = {};
```

Comment:
nit: don't bother assigning a value here, since you'll throw it away in the if/else that follows.  Another pattern you could use is to move the if/else to a function:

```js
try {
  const data = await prepareData(ids, expand === 1);
  res.json(data);
} ... 
```

And your function could look like this (untested):

```js
const prepareData = async (ids, shouldExpand) {
  if (shouldExpand) {
      return Promise.all(
        ids.map(async (id) => {
          // obtain the corresponding feed to populate the author's name to the return data.
          const post = await Post.byId(id);
          return {
            id,
            url: `${postsUrl}/${id}`,
            author: post.feed.author,
            title: post.title,
            publishDate: post.published,
          };
        })
      );
  } else
    return ids
      .map((id) => ({
        id,
        url: `${postsUrl}/${id}`,
      }))
  );
};
```